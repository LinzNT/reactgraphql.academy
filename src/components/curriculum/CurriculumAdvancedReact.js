import React from 'react'
import Link from '../navigation/Link'
import { H2Ref } from '../text'
import Section, { curriedToggleNavigateTo } from './CurriculumSection'
import TestingInReactSession from './sessions/TestingInReactSession'
import HoCsRenderPropsStateReducerSession from './sessions/HoCsRenderPropsStateReducerSession'
import TestingIntroSession from './sessions/TestingIntroSession'
import E2ESession from './sessions/E2ESession'
import ReactPerformanceSession from './sessions/ReactPerformanceSession'
import { ADVANCED_REACT } from '../../config/data'
import selectCurriculumLayout from './selectCurriculumLayout'
import { Li } from '../layout/Ul'
import { trainingTime } from '../utils'
import { curriculumCommonPropTypes } from './'

const CurriculumAdvancedReact = ({
  showTitle = true,
  isOpen,
  enableToggle,
  toggleNavigateTo = `/react/curriculum?tab=${ADVANCED_REACT}`,
  showLinkToCurriculum = true,
  layout,
  trainings,
  training,
}) => {
  const toggleNavigateToSection = curriedToggleNavigateTo(toggleNavigateTo)
  const type = ADVANCED_REACT
  const commonProps = {
    enableToggle,
    toggleNavigateTo: toggleNavigateToSection,
    type,
    isOpen,
  }
  const firstHalf = (
    <React.Fragment>
      <Section
        {...commonProps}
        title="Advanced React Day 1"
        name="day1"
        subTitle="Advanced React patterns, Hooks, and performance"
        trainingTime={trainingTime({ day: 0, training, type })}
        buyPath="/react/training/workshops/advanced-react-patterns-fp-performance"
      >
        <HoCsRenderPropsStateReducerSession title="Functional Programming & Advanced React patterns" />
        <ReactPerformanceSession title="Performance" />
      </Section>
      <Section
        {...commonProps}
        title="Advanced React Day 2"
        name="day2"
        subTitle="Real-world Testing in React"
        trainingTime={trainingTime({ day: 1, training })}
      >
        <TestingIntroSession title="Testing Foundation in JS" />
        <TestingInReactSession title="Testing in React" />
        <E2ESession title="End-to-End Testing" />
      </Section>
    </React.Fragment>
  )

  const title = showTitle ? (
    <H2Ref>
      Advanced React Curriculum{' '}
      <Link to="#curriculum" name="curriculum">
        #
      </Link>
    </H2Ref>
  ) : null

  return selectCurriculumLayout({
    firstHalf,
    title,
    layout,
    type,
    trainings,
    curriculumTo: showLinkToCurriculum ? toggleNavigateTo : undefined,
  })
}

export const TargetAudienceList = () => (
  <React.Fragment>
    <Li>
      Ideal for{' '}
      <strong>React developers with 1+ year of profesional experience</strong>.
      Not for React beginners!
    </Li>
    <Li>
      Get advice to address your complex React challenges with our expert
      coaches.
    </Li>
    <Li>
      Discuss real-world problems with other advanced React developers from
      different companies and countries.
    </Li>
    <Li>
      Take a step forward to become a{' '}
      <strong>Senior React Developer/ Architect</strong> able to make critical
      decisions about the architecture of advanced React applications.
    </Li>
  </React.Fragment>
)

const LearningObjectivesList = () => (
  <React.Fragment>
    <Li>
      Be able to test complex React applications writing tests that promote code
      quality and best programming practices.
    </Li>
    <Li>
      Comprehend modern front-end JavaScript using Functional Programming (FP)
    </Li>
    <Li>
      Understand the best practices and patterns for building real-world
      performant applications using advanced React.
    </Li>
  </React.Fragment>
)

CurriculumAdvancedReact.propTypes = curriculumCommonPropTypes
CurriculumAdvancedReact.LearningObjectivesList = LearningObjectivesList
CurriculumAdvancedReact.TargetAudienceList = TargetAudienceList

export default CurriculumAdvancedReact
