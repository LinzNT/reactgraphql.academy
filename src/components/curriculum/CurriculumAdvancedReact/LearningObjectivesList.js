import React from 'react'
import { Li } from '../../layout/Ul'
import TestingLearningObjectives from '../workshops/CurriculumTestingInReact/LearningObjectivesList'
import StylingLearningObjectives from '../workshops/CurriculumStylingAndAdvUI/LearningObjectivesList'

const LearningObjectivesList = () => (
  <React.Fragment>
    <Li>
      Be able to use advanced React patterns & functional programming to improve
      performance and code quality.
    </Li>
    <StylingLearningObjectives />
    <TestingLearningObjectives />
    <Li>Learn best practices to build real-world React applications.</Li>
  </React.Fragment>
)

export default LearningObjectivesList
