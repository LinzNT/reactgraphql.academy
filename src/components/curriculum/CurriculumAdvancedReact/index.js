import React from 'react'
import { ADVANCED_REACT } from '../../../config/data'
import { sessionsSecondHalf as sessionsSecondHalfBootcamp } from '../CurriculumReactBootcamp'
import Curriculum, { renderSection } from '../Curriculum'

const CurriculumAdvancedReact = ({
  toggleNavigateTo = `/react/curriculum?tab=${ADVANCED_REACT}`,
  training,
  section = {},
  ...rest
}) => {
  const type = ADVANCED_REACT
  const sectionProps = {
    ...section,
    toggleNavigateTo,
    type,
  }
  const renderSectionArgs = {
    training,
    sectionProps,
    titlePrefix: 'Advanced React Day',
  }
  return (
    <Curriculum
      title="Advanced React Curriculum"
      training={training}
      type={type}
      curriculumTo={toggleNavigateTo}
      {...rest}
      firstHalf={sessionsSecondHalfBootcamp.map(
        renderSection(renderSectionArgs)
      )}
    />
  )
}

export default CurriculumAdvancedReact
