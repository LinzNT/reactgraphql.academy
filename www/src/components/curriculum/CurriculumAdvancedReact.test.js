import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import CurriculumAdvancedReact from './CurriculumAdvancedReact'

describe('<CurriculumAdvancedReact />', () => {
  it('should render component', () => {
    const wrapper = shallow(<CurriculumAdvancedReact />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
