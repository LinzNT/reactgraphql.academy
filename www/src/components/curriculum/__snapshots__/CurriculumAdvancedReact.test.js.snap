// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurriculumAdvancedReact /> should render component 1`] = `
<Curriculum
  curriculumTo="/react/curriculum?tab=5ab6244a88546e46fa2b2601full-time"
  firstHalf={
    Array [
      <CurriculumSection
        name="day1"
        subTitle="Advanced React patterns and performance"
        tech="react"
        title="Advanced React Day 1"
        toggleNavigateTo="/react/curriculum?tab=5ab6244a88546e46fa2b2601full-time"
        trainingDateTime="-  09:00 - 18:00"
        trainingId="5ab6244a88546e46fa2b2601"
        trainingInstanceTypeName="full-time"
        trainingType="half-curriculum"
      >
        <React.Fragment>
          <AdvancedReactPatternsSession
            title="Functional Programming & Advanced React patterns"
          />
          <AdvancedHooksAndPerfSession
            title="Advanced Hooks and Performance"
          />
        </React.Fragment>
      </CurriculumSection>,
      <CurriculumSection
        name="day2"
        subTitle="GraphQL 101 & Real-World Testing in React"
        tech="react"
        title="Advanced React Day 2"
        toggleNavigateTo="/react/curriculum?tab=5ab6244a88546e46fa2b2601full-time"
        trainingDateTime="-  09:00 - 18:00"
        trainingId="5ab6244a88546e46fa2b2601"
        trainingInstanceTypeName="full-time"
        trainingType="half-curriculum"
      >
        <React.Fragment>
          <ReactGraphQLIntroSession
            title="GraphQL in React 101"
          />
          <TestingIntroSession
            title="Testing Foundation in JavaScript"
          />
          <TestingInReactSession2
            title="Testing in React & GraphQL"
          />
        </React.Fragment>
      </CurriculumSection>,
      <CurriculumSection
        name="day3"
        subTitle="Building a UI component library"
        tech="react"
        title="Advanced React Day 3"
        toggleNavigateTo="/react/curriculum?tab=5ab6244a88546e46fa2b2601full-time"
        trainingDateTime="-  09:00 - 18:00"
        trainingId="5ab6244a88546e46fa2b2601"
        trainingInstanceTypeName="full-time"
        trainingType="half-curriculum"
      >
        <React.Fragment>
          <StylingInReactSession
            title="Styling in React"
          />
          <DesignSystemSession
            title="Design Systems"
          />
        </React.Fragment>
      </CurriculumSection>,
    ]
  }
  tech="react"
  title="Advanced React Training Outline"
  trainingId="5ab6244a88546e46fa2b2601"
  trainingInstanceTypeName="full-time"
  trainingType="half-curriculum"
/>
`;
